metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py

```diff
@@ -0,0 +1,687 @@
+              EP.VALUE AS table_description
+            FROM sys.tables AS T
+            INNER JOIN sys.extended_properties AS EP
+              ON EP.MAJOR_ID = T.[OBJECT_ID]
+              AND EP.MINOR_ID = 0
+              AND EP.NAME = 'MS_Description'
+              AND EP.CLASS = 1
+            """
+        )
+        for row in table_metadata:
+            self.table_descriptions[
+                f"{db_name}.{row['schema_name']}.{row['table_name']}"
+            ] = row["table_description"]
+
+    def _populate_column_descriptions(self, conn: Connection, db_name: str) -> None:
+        column_metadata = conn.execute(
+            """
+            SELECT
+              SCHEMA_NAME(T.SCHEMA_ID) AS schema_name,
+              T.NAME AS table_name,
+              C.NAME AS column_name ,
+              EP.VALUE AS column_description
+            FROM sys.tables AS T
+            INNER JOIN sys.all_columns AS C
+              ON C.OBJECT_ID = T.[OBJECT_ID]
+            INNER JOIN sys.extended_properties AS EP
+              ON EP.MAJOR_ID = T.[OBJECT_ID]
+              AND EP.MINOR_ID = C.COLUMN_ID
+              AND EP.NAME = 'MS_Description'
+              AND EP.CLASS = 1
+            """
+        )
+        for row in column_metadata:
+            self.column_descriptions[
+                f"{db_name}.{row['schema_name']}.{row['table_name']}.{row['column_name']}"
+            ] = row["column_description"]
+
+    @classmethod
+    def create(cls, config_dict: Dict, ctx: PipelineContext) -> "SQLServerSource":
+        config = SQLServerConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    # override to get table descriptions
+    def get_table_properties(
+        self, inspector: Inspector, schema: str, table: str
+    ) -> Tuple[Optional[str], Dict[str, str], Optional[str]]:
+        description, properties, location_urn = super().get_table_properties(
+            inspector, schema, table
+        )
+        # Update description if available.
+        db_name: str = self.get_db_name(inspector)
+        description = self.table_descriptions.get(
+            f"{db_name}.{schema}.{table}", description
+        )
+        return description, properties, location_urn
+
+    # override to get column descriptions
+    def _get_columns(
+        self, dataset_name: str, inspector: Inspector, schema: str, table: str
+    ) -> List[Dict]:
+        columns: List[Dict] = super()._get_columns(
+            dataset_name, inspector, schema, table
+        )
+        # Update column description if available.
+        db_name: str = self.get_db_name(inspector)
+        for column in columns:
+            description: Optional[str] = self.column_descriptions.get(
+                f"{db_name}.{schema}.{table}.{column['name']}",
+            )
+            if description:
+                column["comment"] = description
+        return columns
+
+    def get_database_level_workunits(  # type: ignore[override]
```

Comment:
why is this a type override?

instead could we use an `assert isinstance(sql_config, SQLServerConfig)` to avoid needing the type ignore?

Reply:
same with below for get_schema_level_workunits

Reply:
I think we don`t need pass sql_config as parameter. We can just take it from self